### PR TITLE
[Sparse] Make functions compatible with PyTorch scalar tensors

### PR DESCRIPTION
**Pull request by BarclayII**
## Description
A sub PR of #5148 .

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: python/dgl/sparse/utils.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+"""Utilities for DGL sparse module."""
```
Comment (frozenbugs) [inline python/dgl/sparse/utils.py:1]: nit: is this sparse specific utils?

File: tests/pytorch/sparse/test_elementwise_op_sp.py [hunk 2]
```diff
@@ -20,7 +20,9 @@ def all_close_sparse(A, row, col, val, shape):
     assert A.shape == shape
 
 
-@pytest.mark.parametrize("v_scalar", [2, 2.5])
+@pytest.mark.parametrize(
+    "v_scalar", [2, 2.5, torch.tensor(2), torch.tensor(2.5)]
```
Comment (mufeili) [inline tests/pytorch/sparse/test_elementwise_op_sp.py:24]: Do we expect to support things like `torch.tensor([2])`?

**Review by mufeili — APPROVED** [review]

File: tests/pytorch/sparse/test_elementwise_op_sp.py (hunk 2 above)
Comment (BarclayII) [reply tests/pytorch/sparse/test_elementwise_op_sp.py:24]: I think so.  It allows us to backpropagate through the scalar coefficient at least.

File: python/dgl/sparse/utils.py (hunk 1 above)
Comment (BarclayII) [reply python/dgl/sparse/utils.py:1]: So far it's sparse specific (really PyTorch specific).  Do you have a better place for it?

**Pull request merged**